syntax = "proto3";
package pkg;




import "model/artifact/package.proto";

/*
Search请求
*/
message SearchRequest {
    /*
    page number
    */
    int32 page = 1 ;
    /*
    size of each page
    */
    int32 pageSize = 2 ;
    /*
    the field need to sort
    */
    string order = 3 ;
    /*
    sdk限制字段使用
    */
    repeated string XXX_RestFieldMask = 4 ;
    /*
    包名称
    */
    string name = 5 ;
    /*
    版本类型 1 程序包,  2 配置包,  4 文件包
    */
    int32 type = 6 ;
    /*
    包分类
    */
    int32 cId = 7 ;
    /*
    包分类标签
    */
    string category = 8 ;
}

/*
Search返回
*/
message SearchResponse {
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated artifact.Package list = 4 ;
}

/*
SearchApi返回
*/
message SearchResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    pkg.SearchResponse data = 4 ;
}

